appserver/deployment/dol/src/main/java/com/sun/enterprise/deployment/annotation/handlers/ResourceHandler.java

```diff
@@ -291,6 +301,24 @@ protected HandlerProcessingResult processResource(AnnotationInfo ainfo,
         return getDefaultProcessedResult();
     }
 
+    /**
+     * Method to verify any other Environment Properties that contain same field name and return value if not empty
+     *
+     * @param fieldName   field to search
+     * @param descriptors available descriptors for EJB
+     * @return String value
+     */
+    public String searchValueIfAvailable(String fieldName, EnvironmentProperty[] descriptors) {
+        Optional<EnvironmentProperty> notEmptyValueOptional = Arrays.stream(descriptors)
```

Comment:
No local variable, no condition:
```java
        return Arrays.stream(descriptors)
                .filter(d -> d.getDisplayName().equalsIgnoreCase(fieldName) ||
                        fieldName.toLowerCase().contains(d.getDisplayName().toLowerCase()))
                .filter(EnvironmentProperty::hasAValue)
                .findFirst()
                .map(EnvironmentProperty::getValue)
                .orElse("");
```
?

Maybe it could be static?

Maybe it could be not public?

Maybe it could be unit-tested?